test/fetch/client-fetch.js

```diff
@@ -165,21 +165,97 @@ test('unsupported formData 1', (t) => {
   })
 })
 
-test('unsupported formData 2', (t) => {
+test('multipart formdata not base64', async (t) => {
+  t.plan(2)
+
+  // Construct example form data, with text and blob fields
+  const formData = new FormData()
+  formData.append('field1', 'value1')
+  const blob = new Blob(['example\ntext file'], { type: 'text/plain' })
+  formData.append('field2', blob, 'file.txt')
+
+  const tempRes = new Response(formData)
+  const boundary = tempRes.headers.get('content-type').split('boundary=')[1]
+  const formRaw = await tempRes.text()
+
+  const server = createServer((req, res) => {
+    res.setHeader('content-type', 'multipart/form-data; boundary=' + boundary)
+    res.write(formRaw)
+    res.end()
+  })
+  t.teardown(server.close.bind(server))
+
+  await new Promise((resolve) => {
+    server.listen(0, async () => {
```

Comment:
I would recommend to avoid promises and callbacks.

Reply:
I rewrote the tests here: https://github.com/nodejs/undici/pull/1606/commits/b079acb0948e32c656f576c7eeadf53db5c5c21b

Let me know if there is anything else that needs doing before we can get this landed